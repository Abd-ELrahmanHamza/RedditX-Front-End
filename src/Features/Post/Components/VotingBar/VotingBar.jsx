import { Container, Up, Count, Down } from "./VotingBar.styled";

import { TbArrowBigTop, TbArrowBigDown } from "react-icons/tb";

import { useState, useEffect } from "react";

import { handleUp, handleDown } from "Features/Post/Utils/upAndDownVoting";

import NumberAbbreviate from "number-abbreviate";

import useFetchFunction from "Hooks/useFetchFunction";

/**
 * VotingBar Component that is in the side of Post Container
 * @returns {React.Component} VotingBar component
 */
const VotingBar = ({ number }) => {
  const [giveUp, setGiveUp] = useState(false);
  const [giveDown, setGiveDown] = useState(false);
  const [count, setCount] = useState(number);
  const [countStr, setCountStr] = useState("");

  const id = "468452313"; // test id

  /**
   * Custom hook
   */
  const [data, error, isLoading, dataFetch] = useFetchFunction();

  useEffect(() => {
    setCountStr(NumberAbbreviate(count, 2));
  }, [count]);

  return (
    <Container>
      <Up
        data-testid="up"
        choosed={giveUp}
        onClick={(event) => {
          event.stopPropagation();
          handleUp(
            giveUp,
            giveDown,
            count,
            setCount,
            setGiveUp,
            setGiveDown,
            id,
            dataFetch
          );
        }}
      >
        <TbArrowBigTop size={22} />
      </Up>
      <Count data-testid="count" up={giveUp} down={giveDown}>
        {countStr}
      </Count>
      <Down
        data-testid="down"
        choosed={giveDown}
        onClick={(event) => {
          event.stopPropagation();
          handleDown(
            giveUp,
            giveDown,
            count,
            setCount,
            setGiveUp,
            setGiveDown,
            id,
            dataFetch
          );
        }}
      >
        <TbArrowBigDown size={22} />
      </Down>
    </Container>
  );
};

export default VotingBar;
