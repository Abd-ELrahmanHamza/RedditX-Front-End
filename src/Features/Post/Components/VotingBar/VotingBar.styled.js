import styled from "styled-components";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: start;
  align-items: center;
  background-color: #cccccc2b;
  padding: 5px;
  width: 7%;
  heigth: 100%;
`;

export const Up = styled.button`
  border: none;
  background-color: transparent;

  svg {
    color: ${(p) => (p.choosed ? "red" : "gray")};
  }

  &:hover {
    background-color: #e1e0e0 !important;

    svg {
      color: red !important;
    }
  }
`;

export const Down = styled.button`
  border: none;
  background-color: transparent;
  color: gray;
  svg {
    color: ${(p) => (p.choosed ? "#0000ff8a" : "gray")};
  }

  &:hover {
    background-color: #e1e0e0 !important;

    svg {
      color: #0000ff8a !important;
    }
  }
`;

export const Count = styled.div`
  color: ${(p) => (p.up ? "red" : p.down ? "#0000ff8a" : "black")};
  font-size: 13px;
  font-weight: 500;
`;
