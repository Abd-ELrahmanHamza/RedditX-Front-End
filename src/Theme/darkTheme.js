/**
 * dark theme colors
 *
 * @type {{id:string}}
 */
const darkTheme = {
  // The id of the theme
  id: "dark",
  // The canvas color is the background color of the website
  canvas: "#030303",
  // The colors of the text on the website
  color: {
    // The primary color of the text on the website
    primary: "#d7dadc",
    // The color of the muted text on the website
    muted: "#7c7c7c",
    // The color of the links on the website
    link: "#0079d3",
  },
  // The colors of the backgrounds in the website
  background: {
    primary: "#1a1a1b",
  },
  // The colors of the lines in the website
  lineColor: {
    primary: "#343536",
  },
};

export default darkTheme;
