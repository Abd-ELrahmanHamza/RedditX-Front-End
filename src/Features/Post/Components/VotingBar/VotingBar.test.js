import { shallow } from "enzyme";
import { screen, render, waitFor } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import darkTheme from "Theme/darkTheme";
import VotingBar from "./VotingBar";

describe("VotingBar Component", () => {
  const wrapper = shallow(<VotingBar number={200} />);

  it("this is a test for VotingBar Component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("this is a test for VotingBar Component 2", () => {
    render(
      <ThemeProvider theme={darkTheme}>
        <VotingBar number={200} />
      </ThemeProvider>
    );
    expect(screen.queryByTestId("count").innerHTML).toEqual("200");
  });

  it("this is a test for VotingBar Component 3", async () => {
    render(
      <ThemeProvider theme={darkTheme}>
        <VotingBar number={200} />
      </ThemeProvider>
    );
    const up = screen.queryByTestId("up");
    up.click();

    await waitFor(() => {
      expect(screen.queryByTestId("count").innerHTML).toEqual("201");
    });
    const down = screen.queryByTestId("down");
    down.click();

    await waitFor(() => {
      expect(screen.queryByTestId("count").innerHTML).toEqual("199");
    });
  });
});
