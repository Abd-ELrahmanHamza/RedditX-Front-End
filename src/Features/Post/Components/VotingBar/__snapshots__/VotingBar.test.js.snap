// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VotingBar Component this is a test for VotingBar Component 1`] = `
<styled.div>
  <styled.button
    choosed={false}
    data-testid="up"
    onClick={[Function]}
  >
    <TbArrowBigTop
      size={22}
    />
  </styled.button>
  <styled.div
    data-testid="count"
    down={false}
    up={false}
  />
  <styled.button
    choosed={false}
    data-testid="down"
    onClick={[Function]}
  >
    <TbArrowBigDown
      size={22}
    />
  </styled.button>
</styled.div>
`;
